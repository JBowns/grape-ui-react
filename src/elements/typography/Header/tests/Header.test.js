import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { Header } from '..';
import 'jest-styled-components';

describe('Header.h1 Component base', () => {
  it('should return a Header object with base styling', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header>Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Header.h1 Component specified styles', () => {
  it('should return a Header object with a base color (implicit)', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header color="green">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a base color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header color="green.base">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a light color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header color="green.light">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a dark color', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header color="green.dark">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with font-size: display', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header display>Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with ellipsis', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header ellipsis>Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a specified font-family', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header fontFamily="monospace">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a specified font-weight', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header fontWeight="bold">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a specified font-weight overriding the display font-weight', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header display fontWeight="bold">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with font-style: italic', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header italic>Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with font-size: lg', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header lg>Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with font-size: sm', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header sm>Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a specified kerning', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header kerning="2px">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a specified text-align', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header textAlign="right">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with a specified text-decoration', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header textDecoration="underline">Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header object with custom margin and padding', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header mb={1} mt={1}>Lorem Ipsum</Header></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Header Element sizes', () => {
  it('should return a Header h1 object', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header.h1>Lorem Ipsum</Header.h1></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header h2 object', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header.h2>Lorem Ipsum</Header.h2></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header h3 object', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header.h3>Lorem Ipsum</Header.h3></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header h4 object', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header.h4>Lorem Ipsum</Header.h4></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header h5 object', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header.h5>Lorem Ipsum</Header.h5></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a Header h6 object', () => {
    const component = renderer.create(<ThemeProvider theme={{}}><Header.h6>Lorem Ipsum</Header.h6></ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
