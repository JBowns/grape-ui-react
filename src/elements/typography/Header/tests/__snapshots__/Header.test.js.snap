// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Element sizes should return a Header h1 object 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header Element sizes should return a Header h2 object 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h2
  className="c0"
>
  Lorem Ipsum
</h2>
`;

exports[`Header Element sizes should return a Header h3 object 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.75rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h3
  className="c0"
>
  Lorem Ipsum
</h3>
`;

exports[`Header Element sizes should return a Header h4 object 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h4
  className="c0"
>
  Lorem Ipsum
</h4>
`;

exports[`Header Element sizes should return a Header h5 object 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1.25rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h5
  className="c0"
>
  Lorem Ipsum
</h5>
`;

exports[`Header Element sizes should return a Header h6 object 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 1rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h6
  className="c0"
>
  Lorem Ipsum
</h6>
`;

exports[`Header.h1 Component base should return a Header object with base styling 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a base color (implicit) 1`] = `
.c0 {
  color: hsl(122.4,39.4%,49.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a base color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,49.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a dark color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,29.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a light color 1`] = `
.c0 {
  color: hsl(122.4,39.4%,69.2%);
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a specified font-family 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: Consolas,"Andale Mono WT","Andale Mono","Lucida Console","Lucida Sans Typewriter","DejaVu Sans Mono","Bitstream Vera Sans Mono","Liberation Mono","Nimbus Mono L",Monaco,"Courier New",Courier,monospace;
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a specified font-weight 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: bold;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a specified font-weight overriding the display font-weight 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 6rem;
  font-weight: 300;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a specified kerning 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a specified text-align 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: right;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with a specified text-decoration 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with font-size: display 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 6rem;
  font-weight: 300;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with font-size: lg 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 3rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with font-size: sm 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;

exports[`Header.h1 Component specified styles should return a Header object with font-style: italic 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-style: italic;
  font-size: 2.5rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin: 0 0 1rem;
}

<h1
  className="c0"
>
  Lorem Ipsum
</h1>
`;
